main.go

```diff
@@ -1,12 +1,29 @@
 package main
 
 import (
+	"context"
+	"flag"
+	"log"
+
 	"github.com/hashicorp/terraform-plugin-sdk/v2/plugin"
 	"github.com/sullivtr/terraform-provider-graphql/graphql"
 )
 
 func main() {
-	plugin.Serve(&plugin.ServeOpts{
-		ProviderFunc: graphql.Provider,
-	})
+	var debugMode bool
```

Comment:
Do you have an example of setting this flag in a terraform context that you can share in this PR. Are you simply adding a `-debug` flag on the `terraform` command?

Reply:
One process to use this with VS Code is:

1. Open the `launch.json` configuration of your provider workspace and add an entry like the following:
    ```
    {
        "version": "0.2.0",
        "configurations": [
            {
                "name": "Debug",
                "type": "go",
                "request": "launch",
                "mode": "debug",
                "program": "${workspaceFolder}",
                "args": [
                    "--debug"
                ],
                "env": {
                    "TF_LOG": "debug"
                }
            }
        ]
    }
    ```
2. Add some breakpoints in the provider source code where you want to investigate
3. Start a debugging session with `F5` and check the output in the debug console, you should see something like:
    ```
    Provider started, to attach Terraform set the TF_REATTACH_PROVIDERS env var:
    
	    TF_REATTACH_PROVIDERS='{"registry.terraform.io/sullivtr/graphql":{"Protocol":"grpc","ProtocolVersion":5,"Pid":714107,"Test":true,"Addr":{"Network":"unix","String":"/tmp/plugin651263617"}}}'
    ```
4. Open a terminal and cd to your terraform configuration directory
5. Set and export the environment variable provided above, e.g.:
    ```
    export TF_REATTACH_PROVIDERS='{"registry.terraform.io/sullivtr/graphql":{"Protocol":"grpc","ProtocolVersion":5,"Pid":714107,"Test":true,"Addr":{"Network":"unix","String":"/tmp/plugin651263617"}}}'
    ```
6. Run terraform in the desired mode, e.g.: `TF_LOG=debug terraform plan`
7. terraform will connect to the existing provider process and break points should kick in

Reply:
Gotcha. Thanks for the detailed explanation. I’ve normally gotten by with simply setting TF_TRACE, or TF_LOG. This is neat and helpful tho.